### PR TITLE
Authorization/ Encryption

### PR DESCRIPTION
**Pull request by HarryFray**
Sorry guys this is a big one. 

- Watch out for the isLoggedIn middle ware I added in the server. you will not be able to use the routes without logging in or temporarily deleting the middle ware. 

- Did not do anything in the DB I think a lot of lines just moved down. Delete my temp DB connect URL did not push that on purpose. 

- The insert New User at the bottom of our server index file should be deleted as the password will not be encrypted through that route. 

- be sure to npm install after all this as I added some packages 

- will go over this in more detail tomorrow let me know if something needs to change

- did my best to leave the front end alone. From what I understand it's about to change a lot

### CODE REVIEW

File: database-mongo/index.js [hunk 1]
```diff
@@ -1,6 +1,6 @@
 const mongoose = require('mongoose');
 
-mongoose.connect('mongodb://me:me@ds241019.mlab.com:41019/greenfield-dev');
+mongoose.connect('mongodb://nn:nn@ds251598.mlab.com:51598/netninja');
```
Comment (anjaliannahuja) [inline database-mongo/index.js:3]: Could we change this line back before pushing?

File: react-client/src/index.jsx [hunk 2]
```diff
@@ -62,40 +66,70 @@ class App extends React.Component {
...
 
+
   handleLoginClick() {
     this.setState({ loginModalTriggered: true, signupModalTriggered: false });
   }
 
   handleSignupClick() {
     this.setState({ signupModalTriggered: true });
   }
+  /*
```
Comment (anjaliannahuja) [inline react-client/src/index.jsx:77]: loving it!

File: database-mongo/config.js [hunk 3]
```diff
@@ -0,0 +1,61 @@
+const passport = require('passport');
```
Comment (anjaliannahuja) [inline database-mongo/config.js:1]: This file looks real as hell 💯 🥇

File: server/index.js [hunk 4]
```diff
@@ -28,46 +75,46 @@ app.get('/', (req, res) => {
   res.sendFile(path.join(`${__dirname}/../react-client/src/index.html`));
 });
 // CATEGORY GET '/api/categories': List of all categories.
-app.get('/api/categories', (req, res) => {
+app.get('/api/categories', isLoggedIn, (req, res) => {
```
Comment (anjaliannahuja) [inline server/index.js:78]: Confused on this for the category requests- do we need the "isLoggedIn"?

File: database-mongo/index.js (hunk 1 above)
Comment (HarryFray) [reply database-mongo/index.js:3]: I can change it before merging

File: server/index.js (hunk 4 above)
Comment (HarryFray) [reply server/index.js:78]: really just depends on if what we want people to have access to and not have access to.

**Pull request closed without merging**